.github/workflows/test.yaml

```diff
@@ -2,6 +2,10 @@ name: Testing
 
 on: [push, pull_request, workflow_dispatch]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
I read the documentation on this but I'm not 100% sure on the effect. At first I thought it would force all these tests to run sequentially, but reading further I think since `github.workflow` refers to this workflow, 'Testing', while `github.ref` is the branch, that this will cancel the old 'Testing' job for a branch when new commits get pushed, which seems sensible. Is that correct?

Reply:
yes that's how it works.